### PR TITLE
Move AnimationState to script_traits

### LINKED ISSUE #9223: Move AnimationState to script_traits
This is defined in msg but used by code that depends on script_traits.

Code: `components/msg/constellation_msg.rs`, `components/script_traits/lib.rs`

### PR DESCRIPTION
**Pull request by shssoichiro**
Resolves #9223

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9295)

### CODE REVIEW

File: components/layout/animation.rs [hunk 1]
```diff
@@ -7,7 +7,8 @@
 use flow::{self, Flow};
 use gfx::display_list::OpaqueNode;
 use incremental::RestyleDamage;
-use msg::constellation_msg::{AnimationState, ConstellationChan, PipelineId};
+use msg::constellation_msg::{ConstellationChan, PipelineId};
+use script_traits::AnimationState;
```
Comment (KiChjang) [inline components/layout/animation.rs]: nit: Meld into next line

File: components/script_traits/script_msg.rs [hunk 2]
```diff
@@ -9,11 +9,12 @@ use canvas_traits::CanvasMsg;
 use euclid::point::Point2D;
 use euclid::size::Size2D;
 use ipc_channel::ipc::IpcSender;
-use msg::constellation_msg::{AnimationState, NavigationDirection};
+use msg::constellation_msg::{NavigationDirection};
```
Comment (KiChjang) [inline components/script_traits/script_msg.rs]: nit: Remove the braces.

File: components/script_traits/script_msg.rs [hunk 3]
```diff
@@ -9,11 +9,12 @@ use canvas_traits::CanvasMsg;
...
 use msg::constellation_msg::{Failure, MozBrowserEvent, PipelineId};
 use msg::constellation_msg::{LoadData, MouseButton, SubpageId};
 use offscreen_gl_context::GLContextAttributes;
 use style_traits::viewport::ViewportConstraints;
+use super::AnimationState;
```
Comment (KiChjang) [inline components/script_traits/script_msg.rs]: I believe `use AnimationState;` should be enough.

File: components/compositing/constellation.rs [hunk 4]
```diff
@@ -43,6 +42,7 @@ use pipeline::{CompositionPipeline, InitialPipelineState, Pipeline, Unprivileged
 use profile_traits::mem;
 use profile_traits::time;
 use sandboxing;
+use script_traits::AnimationState;
```
Comment (KiChjang) [inline components/compositing/constellation.rs]: nit: Meld into next line.

File: components/compositing/compositor_thread.rs [hunk 5]
```diff
@@ -14,11 +14,11 @@ use ipc_channel::ipc::{self, IpcReceiver, IpcSender};
 use layers::layers::{BufferRequest, LayerBufferSet};
 use layers::platform::surface::{NativeDisplay, NativeSurface};
 use msg::compositor_msg::FrameTreeId;
-use msg::constellation_msg::{AnimationState, PipelineId};
+use msg::constellation_msg::PipelineId;
```
Comment (KiChjang) [inline components/compositing/compositor_thread.rs]: nit: Meld into next line.

**Comment by jdm** [comment]
Travis says that `./mach test-tidy` reports a couple style nits:
```
./components/script/dom/document.rs:84: use statement is not in alphabetical order
	expected: msg::constellation_msg::{ALT, CONTROL, SHIFT, SUPER}
	found: msg::constellation_msg::PipelineId
./components/script_traits/script_msg.rs:13: use statement is not in alphabetical order
	expected: msg::constellation_msg::{NavigationDirection}
	found: msg::constellation_msg::{Failure, MozBrowserEvent, PipelineId}
```

**Comment by shssoichiro** [comment]
I'll let you know. I think I fubar'd my branch (locally). Let me try this again.

"I love rebasing." -Nobody ever

**Comment by KiChjang** [comment]
Here's a link to our wiki that would hopefully help you out: https://github.com/servo/servo/wiki/Github-workflow

**Comment by shssoichiro** [comment]
Success! Thanks for the link, that was helpful. I messed up originally by not using `git push -f` and when I did `git pull` like git told me to, it messed up my local branch.

**Comment by KiChjang** [comment]
Glad you sorted stuff out!

@bors-servo r+

**Comment by KiChjang** [comment]
Whoops, looks like other PRs broke your changes! You'll unfortunately need to rebase again :(

**Comment by shssoichiro** [comment]
It was easier this time around, now that I know what I'm doing. :)

**Comment by KiChjang** [comment]
@bors-servo retry
* #7846

**Pull request merged**